packages/webvh/src/dids/WebVhDidRegistrar.ts

```diff
@@ -0,0 +1,218 @@
+import type {
+  AgentContext,
+  DidCreateOptions,
+  DidCreateResult,
+  DidDeactivateOptions,
+  DidDeactivateResult,
+  DidRegistrar,
+  DidUpdateOptions,
+  DidUpdateResult,
+  VerificationMethod,
+} from '@credo-ts/core'
+
+import { DidDocument, DidDocumentRole, DidRecord, DidRepository, JsonTransformer, Kms } from '@credo-ts/core'
+import { type DIDLog, MultibaseEncoding, createDID, multibaseEncode, updateDID } from 'didwebvh-ts'
+
+import { WebvhDidCrypto } from './WebvhDidCrypto'
+import { WebvhDidCryptoSigner } from './WebvhDidCryptoSigner'
+
+interface WebVhDidCreateOptions extends DidCreateOptions {
+  domain: string
+  path?: string
+}
+
+interface WebVhDidUpdateOptions extends DidUpdateOptions {
+  didDocument: DidDocument
+}
+
+const normalizeMethodArray = (arr?: (string | { id: string })[]) =>
+  arr?.map((item) => (typeof item === 'string' ? item : item.id))
+
+/**
+ * DID Registrar implementation for the 'webvh' method.
+ * Handles creation, update, and (future) deactivation of DIDs using the webvh method.
+ */
+export class WebVhDidRegistrar implements DidRegistrar {
+  public readonly supportedMethods: string[] = ['webvh']
+
+  /**
+   * Creates a new DID document and saves it in the repository.
+   * Handles crypto setup, DID generation, and persistence.
+   * The `paths` option (string with `/` or array of segments) allows adding sub-identifiers
+   * after the domain, joined with `:` in the resulting DID.
+   * @param agentContext The agent context.
+   * @param options The creation options, including domain, optional paths, endpoints, controller, signer, and verifier.
+   * @returns The result of the DID creation, with error handling.
+   */
+  public async create(agentContext: AgentContext, options: WebVhDidCreateOptions): Promise<DidCreateResult> {
+    try {
+      const { domain, path } = options
+      const paths = path?.replace(/^\/|\/$/g, '').split('/')
+      const domainKey = paths?.length ? [domain, ...paths].join(':') : domain
+      const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+      const record = await didRepository.findSingleByQuery(agentContext, {
+        role: DidDocumentRole.Created,
+        domain: domainKey,
+        method: 'webvh',
+      })
+      if (record) return this.handleError(`A record with domain "${domain}" already exists.`)
+
+      // Create crypto instance
+      const { publicKeyMultibase, keyId } = await this.generatePublicKey(agentContext)
+      const signer = new WebvhDidCryptoSigner(agentContext, publicKeyMultibase, keyId)
+      const verifier = new WebvhDidCrypto(agentContext)
+      const baseDid = `did:webvh:{SCID}:${domain}`
+
+      // Create DID
+      const { did, doc, log } = await createDID({
+        domain,
+        paths,
+        signer,
+        updateKeys: [publicKeyMultibase],
+        verificationMethods: [
+          {
+            controller: baseDid,
+            type: 'Ed25519VerificationKey2020',
```

Comment:
```suggestion
            type: 'Multikey',
```

If this key is intended to sign the attested resources with eddsa-jcs-2022, I would recommend using the `Multikey` type to favor interop as per the specification:
https://www.w3.org/TR/vc-di-eddsa/#multikey